jerry-core/parser/js/js-parser.c

```diff
@@ -2396,6 +2396,11 @@ parser_parse_source (const uint8_t *arg_list_p, /**< function argument list */
   context.last_statement.current_p = NULL;
   context.status_flags |= parse_opts & PARSER_STRICT_MODE_MASK;
 
+  if (parse_opts & ECMA_PARSE_EVAL)
```

Comment:
AFAIK there should be a `ENABLED (JERRY_ES2015_MODULE_SYSTEM)` check around this.